core/src/main/java/org/apache/calcite/plan/SubstitutionVisitor.java

```diff
@@ -1407,14 +1406,10 @@ protected boolean isWeaker(MutableRel rel0, MutableRel rel) {
       return false;
     }
 
-    RexExecutorImpl rexImpl =
-        (RexExecutorImpl) (rel.cluster.getPlanner().getExecutor());
-    RexImplicationChecker rexImplicationChecker =
```

Comment:
Instead of removing RexImplicationChecker and hiding it behind the RexExecutor, it could be better to improve RexImplicationChecker to work with a RexExecutor and not only with a RexExecutorImpl.